src/rofiles-fuse/main.c

```diff
@@ -385,7 +385,12 @@ do_open (const char *path, mode_t mode, struct fuse_file_info *finfo)
       if (opt_copyup)
         {
           (void) close (fd);
-          fd = openat (basefd, path, finfo->flags & ~O_TRUNC, mode);
+          /* Note that unlike the initial open, we will pass through
+           * O_TRUNC.  More ideally in this copyup case we'd avoid copying
```

Comment:
Hmm, so then we don't need to `ftruncate` down below again, right? Shouldn't the `if` condition at 399 be changed to

```
if (!opt_copyup && (finfo->flags & O_TRUNC))
```

?